hibernate-query-interpreter/src/main/antlr/org/hibernate/query/parser/internal/hql/antlr/HqlParser.g4

```diff
@@ -172,7 +176,7 @@ dynamicInstantiationArgs
 	;
 
 dynamicInstantiationArg
-	:	dynamicInstantiationArgExpression (asKeyword? IDENTIFIER)?
+	:	dynamicInstantiationArgExpression (asKeyword? alias)?
```

Comment:
This usage is in fact declaring an alias, but its intent is not to be reusable in other parts of the query.  The intent for being able to alias dynamic instantiation arguments is to potentially allow these to name JavaBean-style property names for setter injection rather than needing to use ctors strictly.